packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -211,14 +215,22 @@ export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: S
     return result
   }
 
+  function ignoreConfigurationOptions(initConfiguration: C, options: readonly string[]) {
+    for (const option in initConfiguration) {
+      if (includes(options, option)) {
+        delete initConfiguration[option]
+      }
+    }
```

Comment:
NIT:
```suggestion
    for (const option of options) {
      delete initConfiguration[option]
    }
```

Reply:
If I do this, typescript complains `No index signature with a parameter of type 'string' was found on type RumInitConfiguration.ts`.
I found 4 options:
- `delete initConfiguration[option as keyof RumInitConfiguration]`: IMO it adds confusion as the key I'm trying to remove are not part of RumInitConfiguration
- `delete initConfiguration[option as any]`: I generally avoid it as it is error prone
- Ignore typescript check: Not a big fan, either
- Change the loop: What I did

But I've no strong opinion about it. 
Is there a better solution? or what do you prefer?

Reply:
what about 
```ts
ignoredConfigurationOptions.forEach((option) => {
  delete (initConfiguration as InitConfiguration)[option]
})
```

Reply:
I think a TS cast is fine here. `any` should be fine IMO. I think it's better than iterating over all configuration options. We may imagine more complex typings, but I don't think it's worth it here, and I think we'll need an imprecise cast no matter what.

Reply:
I go with the cast of Bastien. It's short and clear :)